### PR TITLE
Drop HAVE_CXX11 macro, fix for #4284

### PR DESCRIPTION
**Pull request by shiyi001**
Drop HAVE_CXX11 macro. Some fix for #4284 on gpl submodule has been submitted [there](https://github.com/shogun-toolbox/shogun-gpl/pull/9).

### CODE REVIEW

File: src/shogun/lib/config.h.in [hunk 1]
```diff
@@ -95,8 +95,6 @@
 #cmakedefine HAVE_ALIGNED_MALLOC 1
 #cmakedefine HAVE_POSIX_MEMALIGN 1
 
-#cmakedefine HAVE_CXX11 1
```
Comment (vinx13) [inline src/shogun/lib/config.h.in]: have you removed all HAVE_CXX11 macros? otherwise this will break other things

Comment (shiyi001) [reply src/shogun/lib/config.h.in]: Yes. I have removed all the HAVE_CXX11 macros including the gpl submodule.

**Comment by sighingnow** [comment]
This PR seems not the the trigger of the failure `LibLinearRegression.lr_without_bias` test on MacOS. All this PR do is removing dead code. The failure exists before this PR, may be a separate issue is needed to address that case.

**Comment by karlnapf** [comment]
Thanks for the patch!

**Pull request merged**